### ISSUE TITLE
Auto-Tagging / Insert Meta Data / Textual information chunks into PNG

### ISSUE THREAD

**Issue by avviano** [chaiNNer-org/chaiNNer#2197]
When uploading images to micro stock sites or to simply stay organized and improve search capabilities, an important and tedious process is tagging images appropriately. This process takes a lot of work and time and includes meta like title, image description and keywords. With AI, this process can be automated (some project to do that are available on Github) 

It would be great if Chainner could fill the gap and offer nodes to make it possible to interrograte an image for title, description and keywords and write this text information meta data / textual information chunks into a PNG file. 

Basically we would need 

1.) one node to interrogate images using AI to text meta like title, description and keywords. Title could be taken from first x-number words in description.    
2.) one node for writing or appending textual information chunks such as title, description, keywords, author, copyright, creation time, image source and comment into the PNG file. 

A node to connect to Stable Diffusion to interrogate images for a text description already exists. But it seems there is no node to write that text into PNG meta. 

If we had a node to interrograte for title and keywords as well and then also had nodes for all three (title, description and keywords) to write this information into the PNG meta text chunks, that would be amazing.

Basically a feature request for the following nodes: 

1.) Auto-Keyword Generation based on image interrogration by AI model (or better title, description and keywords). This could be achieved either through: 

  a.)  Use an existing classification model or train a custom one based on available datasets using a machine learning framework like TensorFlow.  See examples on git such as torresflo/Tag-Machine or vladmandic/pigallery (or search AI Image Auto-Tagging on git)

  b.) Use a pre-trained neural network like ResNet or Inception that's been trained on millions of images. Extract features from the image using the network and compare to features of tagged images to find similarity and suggest relevant tags.

  c.) API calls to existing online service such as Everypixel ( https://labs.everypixel.com/keywording ), Google Vision or AWS Rekognition. The returned labels for detected image objects, scenes and faces etc. could then be used as keyword tags. These APIs also provide confidence scores to determine the most relevant tags. The most relevant tags can then be selected automatically based on confidence score thresholds.

2.) Write Metadata to PNG
Title, Author, Description, Copyright, Creation Time, Source, Comment and Keywords
[png-metadata](https://github.com/hometlt/png-metadata) or search Git for "Write - Read PNG Meta)

Cheers
